File: leetcode/silding_window/643/imp.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        if len(nums) < k or k == 0:
            return 0.0

        sum = 0.0
        maxSum = 0.0

        for i in range(k):
            sum += nums[i]
        
        maxSum = sum
        
        for i in range(k, len(nums)):
            sum += nums[i]
            sum -= nums[i - k]

            if sum > maxSum:
                maxSum = sum

        return maxSum / k

File: leetcode/silding_window/643/imp.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        if (k <= 0 || k > nums.size()) {
            return 0.0;
        }

        double sum = 0;
        double maxSum = 0;

        for (int i = 0; i < k; ++i) {
            sum += nums[i];
        }
        maxSum = sum;

        for (int i = k; i < nums.size(); ++i) {
            sum += nums[i];
            sum -= nums[i - k];

            if (sum > maxSum) {
                maxSum = sum;
            }
        }

        return maxSum / k;
    }
};